crates/primitives/state_update/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+mod into_starknet_core;
```

Comment:
We use our own primitives for all data in the database because we use bincode to serialize the data. The serialization methods of external crates do not allow us to serialize/deserialize using bincode. Using our own primitives also enables us to implement all our methods without needing to create traits or conversion implementations.